app/code/community/Doofinder/Feed/Model/Generator.php

```diff
@@ -773,4 +787,40 @@ protected function _cleanFieldValue($field)
 
         return preg_replace($valid_utf8, '$1', $field);
     }
+
+    /**
+     * Check if array is flat (not multidimensional)
+     *
+     * @param array $arr
+     * @return boolean
+     */
+    protected function _isArrayFlat(array $arr)
+    {
+        $isFlat = true;
+
+        foreach ($arr as $item)
+        {
+            if (is_array($item))
+            {
+                $isFlat = false;
+                break;
+            }
+        }
+
+        return $isFlat;
+    }
+
+    /**
+     * Flatten array recursively
+     *
+     * @notice This requires PHP5.3+
+     *
+     * @param array @arr
+     * @return array
+     */
+    protected function _flattenArray(array $arr) {
+        $flattenedArray = array();
+        array_walk_recursive($arr, function($item) use (&$flattenedArray) { $flattenedArray[] = $item; });
```

Comment:
Is this syntax compatible with PHP 5.3? I think it isn't... lambda functions are not supported until 5.4... Could you refactor this to make it compatible with older versions of PHP?